analysis_plan.md

```diff
@@ -0,0 +1,27 @@
+## Analysis Plan
+
+# Plan 1
+I am interested in comparing kmers from the sequences from Africa to the sequences from South America. I am interested in looking at the similarity of these sequences to each other and to themselves over time, predicting that as a whole sequences from Africa at the beginning of the pandemic in late 2019 would be more similar to sequences from Africa later in the pandemic than to South American sequences  overall due to the big differences in geographic location. I believe I can make use of kmers to do this, based on some information from https://bioinfologics.github.io/post/2018/09/17/k-mer-counting-part-i-introduction/. I can take kmers from the South America sequences and see if there are more of the same kmers present in the rest of the South America data or the Africa sequences. The more kmers that are shared between the sets of data, the more similar they are. I can do this for varying kmer lengths; longer lengths will indicate more similarity, but also risk being too specific to show potential similarities in smaller sections of sequence. The article suggests a maximim length of around 31 because that is still a manageable number for Int64 to handle. The functions I can make use of are:
+
+- The parse_fasta and parse_header functions, in order to traverse the data from the file
+- The kmer composition one from assignment 4
+- A function that can compare the sequences from each kmer across groups of data, keeping track of how many kmers early Africa and late Africa data have in common and also how many kmers early Africa and South America have in common
+
+I could present this information using a line plot that records the size of the kmer against the amount of kmers in common for early Africa against late Africa and early Africa against South America. The lines may vary as the kmer sizes do, but lines that are higher up on the graph will obviously indicate more kmers in common and therefore may be indicators that these two groups of data are more similar than the other.

```

Comment:
I'm not totally clear on what this will look like - you're going to get kmers of a bunch of different lengths and do a separate line for each? Ad what's `x` and `y`? Are you going to pool the kmers from all of the sequences in each category? I think you may need to think through this a bit more and maybe sketch out exactly what data is going to go into the plot

Reply:
Also, if you're going to do a bunch of different kmer lengths, be mindful of computing time. Be sure to benchmark how long the functions will take to run on a small number of sequences before you try to scale up to the whole dataset